### PR TITLE
add 4 new rules for ReactJS ESLint configuration

### PR DESCRIPTION
**Pull request by fpetitit**
Ref http://github.com/lrqdo/front-web/issues/4129

### CODE REVIEW

File: javascript/linters/.eslintrc_react [hunk 1]
```diff
@@ -12,6 +12,10 @@
      */
     "react/display-name": 0,
     "react/jsx-boolean-value": 2,
+    "react/jsx-closing-bracket-location": [2, "line-aligned"],
+    "react/jsx-first-prop-new-line": [2, "multiline"],
+    "react/jsx-indent-props": [2, 4],
```
Comment (jbcochery) [inline javascript/linters/.eslintrc_react:24]: Does that mean 2 spaces are allowed ?

Comment (rik) [inline javascript/linters/.eslintrc_react:24]: 2 means error

**Comment by rik** [comment]
👍 and thanks for starting the conversation in an issue before submitting the PR.

**Comment by jbcochery** [comment]
👍

**Comment by bbecquet** [comment]
:+1: Eager to see all these new red and orange underlines in my code :smile:

**Comment by fpetitit** [comment]
Sorry to make it in the same PR but I forgot that one of the rules is only available since eslint-plugin-react 5, which set React version 15 by default.
We must now specify our React version if lower than 15.

**Comment by fpetitit** [comment]
And we must upgrade our eslint-plugin-react to ^5.0.0, which is defined into front-web.

File: javascript/linters/.eslintrc_react [hunk 2]
```diff
@@ -3,6 +3,14 @@
   "plugins": [
     "react"
   ],
+  {
+    "settings": {
+      "react": {
+        "pragma": "React",  // Pragma to use, default to "React"
```
Comment (rik) [inline javascript/linters/.eslintrc_react]: If it defaults to React, why include it? What other pragmas are they?

File: javascript/linters/.eslintrc_react [hunk 3]
```diff
@@ -3,6 +3,14 @@
...
+        "version": "0.14" // React version, default to the latest React stable release
```
Comment (rik) [inline javascript/linters/.eslintrc_react]: No need for the comment that was copy/pasted from the docs ;)

Comment (fpetitit) [inline javascript/linters/.eslintrc_react]: Idem, I thought that it should be useful to keep them.
No problem if you want to remove comments and default values of course.

Comment (rik) [inline javascript/linters/.eslintrc_react]: Idem ?

**Pull request merged**